pkg/proxy/proxy.go

```diff
@@ -0,0 +1,134 @@
+package proxy
+
+// This file contains this project's custom code, as opposed to kubectl.go
+// which contains code retrieved from the kubernetes project.
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func singleJoiningSlash(a, b string) string {
+	aslash := strings.HasSuffix(a, "/")
+	bslash := strings.HasPrefix(b, "/")
+	switch {
+	case aslash && bslash:
+		return a + b[1:]
+	case !aslash && !bslash:
+		return a + "/" + b
+	}
+	return a + b
+}
+
+func injectOwnerReference(h http.Handler) http.Handler {
+	logrus.Info("injecting owner reference")
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == http.MethodPost {
+			dump, _ := httputil.DumpRequest(req, false)
+			fmt.Println(string(dump))
+
+			user, _, ok := req.BasicAuth()
+			if !ok {
+				logrus.Error("basic auth header not found")
```

Comment:
I think that it might be better to error out, The operator will try again fairly soon and we are in an eventually consistent model. 

I think we just may need to log the correct error, and that I think means the ansible modules give the right error and then the status piece picks up and displays that error IMO

Reply:
And I think now's the time to be strict, and we'll learn through alpha and beta phases if being this strict creates problems.